source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/form8/Form8PaymentItemsSubForm.tsx

```diff
@@ -0,0 +1,205 @@
+    setAmountFields(index, isGstRequired, values.paymentItems[index].pretaxAmount);
+  };
+
+  const onPretaxAmountUpdated = (index: number, newValue: string): void => {
+    const isGstRequired = stringToBoolean(values.paymentItems[index].isGstRequired);
+    const cleanValue = getCurrencyCleanValue(newValue);
+
+    setAmountFields(index, isGstRequired, cleanValue);
+  };
+
+  const onTaxAmountUpdated = (index: number, newValue: string): void => {
+    let totalAmount = 0;
+    const preTaxAmount = values.paymentItems[index].pretaxAmount;
+    const cleanValue = getCurrencyCleanValue(newValue);
+
+    totalAmount = preTaxAmount + cleanValue;
+    setFieldValue(`paymentItems[${index}].totalAmount`, totalAmount);
+  };
+
+  const setAmountFields = (index: number, gstRequired: boolean, pretaxAmount: number): void => {
+    let totalAmount = 0;
+    let taxAmount = 0;
+
+    if (gstRequired) {
+      taxAmount = pretaxAmount * gstConstantPercentage;
+      totalAmount = taxAmount + pretaxAmount;
+    } else {
+      totalAmount = pretaxAmount;
+    }
+
+    setFieldValue(`paymentItems[${index}].taxAmount`, taxAmount);
+    setFieldValue(`paymentItems[${index}].totalAmount`, totalAmount);
+  };
+
+  return (
+    <FieldArray
+      name="paymentItems"
+      render={arrayHelpers => {
+        return (
+          <>
+            {values.paymentItems.map((item, index) => (
+              <div data-testid={`paymentItems[${index}]`}>
+                <StyledSubHeader>
+                  <label>Payment Item {index + 1}</label>
+                  <StyledRemoveLinkButton
+                    title="Delete Payment Item"
+                    data-testid={`paymentItems[${index}].delete-button`}
+                    variant="light"
+                    onClick={() => {
+                      setRowToDelete(index);
+                      setShowModal(true);
+                    }}
+                  >
+                    <FaTrash size="2rem" />
+                  </StyledRemoveLinkButton>
+                </StyledSubHeader>
+
+                <SectionField label="Item" labelWidth="4" required>
+                  <Select
+                    field={`paymentItems[${index}].paymentItemTypeCode`}
+                    options={paymentItemTypesOptions}
+                    placeholder="Select..."
+                    required
+                  />
+                </SectionField>
+                <SectionField label="Payment (w/o GST)" required>
+                  <FastCurrencyInput
+                    allowNegative
+                    formikProps={formikProps}
+                    field={`paymentItems[${index}].pretaxAmount`}
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                      onPretaxAmountUpdated(index, e.target.value);
+                    }}
+                    onBlurChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                      // activitiesUpdated();
```

Comment:
remove commented code and logs when this is no longer in draft.

Reply:
updated